### PR TITLE
rebase_second_snapshot_to_base: support luks

### PR DESCRIPTION
**Pull request by lolyu**
1. support luks as the base image
2. replace functions from `qemu_disk_img` with those in
`provider/qemu_img_utils`.

Signed-off-by: lolyu <lolyu@redhat.com>

### CODE REVIEW

**Comment by lolyu** [comment]
depends on #https://github.com/avocado-framework/avocado-vt/pull/2288

**Comment by lolyu** [comment]
@hellohellenmao @YongxueHong Please take a review.

File: qemu/tests/rebase_second_snapshot_to_base.py [hunk 1]
```diff
@@ -27,82 +28,85 @@ def run(test, params, env):
...
-        img = QemuImg(img_param, data_dir.get_data_dir(), tag)
-        return img, img_param
-
-    def _get_compat_version():
-        """Get snapshot compat version."""
-        if params.get("image_extra_params") is None:
-            # default compat version for now is 1.1
-            return "1.1"
-        return params.get("image_extra_params").split("=")[1]
+    def get_img_objs(images, params):
```
Comment (hereischen) [inline qemu/tests/rebase_second_snapshot_to_base.py]: nice one

**Comment by lolyu** [comment]
```
 (1/8) Host_RHEL.m8.u1.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.1.x86_64.io-github-autotest-qemu.rebase_second_snapshot_to_base.cache_mode.off: PASS (335.40 s)
 (2/8) Host_RHEL.m8.u1.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.1.x86_64.io-github-autotest-qemu.rebase_second_snapshot_to_base.cache_mode.writethrough: PASS (369.29 s)
 (3/8) Host_RHEL.m8.u1.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.1.x86_64.io-github-autotest-qemu.rebase_second_snapshot_to_base.cache_mode.writeback: PASS (337.49 s)
 (4/8) Host_RHEL.m8.u1.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.1.x86_64.io-github-autotest-qemu.rebase_second_snapshot_to_base.cache_mode.none: PASS (345.60 s)
 (5/8) Host_RHEL.m8.u1.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.1.x86_64.io-github-autotest-qemu.rebase_second_snapshot_to_base.cache_mode.unsafe: PASS (336.07 s)
 (6/8) Host_RHEL.m8.u1.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.1.x86_64.io-github-autotest-qemu.rebase_second_snapshot_to_base.cache_mode.directsync: PASS (391.48 s)
 (7/8) Host_RHEL.m8.u1.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.1.x86_64.io-github-autotest-qemu.rebase_second_snapshot_to_base.compat_level.compat_0.10: PASS (342.42 s)
 (8/8) Host_RHEL.m8.u1.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.1.x86_64.io-github-autotest-qemu.rebase_second_snapshot_to_base.compat_level.compat_default: PASS (337.90 s)
```

**Comment by lolyu** [comment]
depends on #https://github.com/avocado-framework/avocado-vt/pull/2288

**Comment by lolyu** [comment]
```
 (1/8) Host_RHEL.m8.u1.luks.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.rebase_second_snapshot_to_base.cache_mode.off: PASS (760.23 s)
 (2/8) Host_RHEL.m8.u1.luks.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.rebase_second_snapshot_to_base.cache_mode.writethrough: PASS (1139.68 s)
 (3/8) Host_RHEL.m8.u1.luks.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.rebase_second_snapshot_to_base.cache_mode.writeback: PASS (932.31 s)
 (4/8) Host_RHEL.m8.u1.luks.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.rebase_second_snapshot_to_base.cache_mode.none: PASS (858.80 s)
 (5/8) Host_RHEL.m8.u1.luks.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.rebase_second_snapshot_to_base.cache_mode.unsafe: PASS (871.31 s)
 (6/8) Host_RHEL.m8.u1.luks.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.rebase_second_snapshot_to_base.cache_mode.directsync: PASS (1082.33 s)
 (7/8) Host_RHEL.m8.u1.luks.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.rebase_second_snapshot_to_base.compat_level.compat_0.10: PASS (979.82 s)
 (8/8) Host_RHEL.m8.u1.luks.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.rebase_second_snapshot_to_base.compat_level.compat_default: PASS (872.94 s)
```

**Comment by hellohellenmao** [comment]
Test for Windows - Pass. 
(1/8) Host_RHEL.m8.u1.luks.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.rebase_second_snapshot_to_base.cache_mode.off: PASS (242.32 s)
 (2/8) Host_RHEL.m8.u1.luks.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.rebase_second_snapshot_to_base.cache_mode.writethrough: PASS (248.56 s)
 (3/8) Host_RHEL.m8.u1.luks.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.rebase_second_snapshot_to_base.cache_mode.writeback: PASS (242.88 s)
 (4/8) Host_RHEL.m8.u1.luks.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.rebase_second_snapshot_to_base.cache_mode.none: PASS (244.20 s)
 (5/8) Host_RHEL.m8.u1.luks.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.rebase_second_snapshot_to_base.cache_mode.unsafe: PASS (240.09 s)
 (6/8) Host_RHEL.m8.u1.luks.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.rebase_second_snapshot_to_base.cache_mode.directsync: PASS (260.35 s)
 (7/8) Host_RHEL.m8.u1.luks.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.rebase_second_snapshot_to_base.compat_level.compat_0.10: PASS (244.67 s)
 (8/8) Host_RHEL.m8.u1.luks.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.rebase_second_snapshot_to_base.compat_level.compat_default: PASS (245.91 s)

**Comment by lolyu** [comment]
@YongxueHong please help to review, thanks

**Pull request merged**